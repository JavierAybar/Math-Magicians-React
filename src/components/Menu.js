import { NavLink } from 'react-router-dom';
import '../styles/Menu.css';
// eslint-disable-next-line arrow-body-style
const Menu = () => {
  return (
    <nav className="menu-container">
      <h1 className="menu-title">Math Magicians 🧙‍♂️</h1>
      <ul className="menu-links-container">
        <li><NavLink end className={({ isActive }) => (isActive ? 'activeLink' : 'desactiveLink')} to="/">Home</NavLink></li>
        <li><NavLink end className={({ isActive }) => (isActive ? 'activeLink' : 'desactiveLink')} to="/calculate">Calculator</NavLink></li>
        <li><NavLink end className={({ isActive }) => (isActive ? 'activeLink' : 'desactiveLink')} to="/quotes">Quotes</NavLink></li>
      </ul>
    </nav>
  );
};

export default Menu;
