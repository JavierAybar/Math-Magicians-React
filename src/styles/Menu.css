li {
  list-style: none;
  cursor: pointer;
}

a {
  font-family: math, sans-serif;
}

.menu-container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: #222;
  padding: 10px;
}

.menu-title {
  margin-left: 50px;
  color: #479a47;
  font-size: 2rem;
  font-family: math, sans-serif;
}

.menu-links-container {
  display: flex;
  gap: 35px;
  margin-right: 200px;
  font-size: 1.1rem;
}

.activeLink {
  color: #2acd2a;
}

.desactiveLink {
  color: #0f8a3e;
  text-decoration: none;
}
