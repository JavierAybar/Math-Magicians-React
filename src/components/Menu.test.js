import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import Menu from './Menu';

test('renders correctly', () => {
  const { container } = render(
    <Router>
      <Menu />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
