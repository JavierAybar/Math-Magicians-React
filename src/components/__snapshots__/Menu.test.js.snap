// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    class="menu-container"
  >
    <h1
      class="menu-title"
    >
      Math Magicians üßô‚Äç‚ôÇÔ∏è
    </h1>
    <ul
      class="menu-links-container"
    >
      <li>
        <a
          aria-current="page"
          class="activeLink"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="desactiveLink"
          href="/calculate"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="desactiveLink"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
